### PR TITLE
MAINT: Remove Selenium dependency and harmonize storage locations

### PR DESCRIPTION
**Pull request by pastewka**
This PR will address:
- Selenium dependency, #787 
- Storage locations, #801

### CODE REVIEW

**Comment by pastewka** [comment]
@mcrot - I've run into a problem and need your advise here:

Both `Topography` and `Analysis` models now have a `storage_prefix` attribute that defines the prefix in the storage system. It need to know the `id` of the model to generate the prefix. The `id` is only known once the model has been saved to the database. This cause problems in the tests, where the model is initialized with some data file and at this point the `id` is not yet known. The solutions to this that I've found are:

- Switch the `id` field to a UUID (see here: https://stackoverflow.com/questions/41428290/django-instance-id-none-when-uploading-image).
- Overload the `save` method (see here: https://stackoverflow.com/questions/9968532/django-admin-file-upload-with-current-model-id)

I'd actually be in favor of using UUIDs for primary keys as this may have security advantages (you cannot guess an id). It may also help here: #307 . However switching to UUIDs would require a larger migration of the database.

**Comment by pastewka** [comment]
I've implemented option 2 and that seems to work

**Comment by pastewka** [comment]
Hi @mcrot - this is ready for your review

File: topobank/manager/signals.py [hunk 1]
```diff
@@ -47,29 +47,46 @@ def delete_datafile(datafile_attr_name):
...
+        for filename in filenames:
+            _log.info(f'Deleting file {path}/{filename}...')
+            default_storage.delete(f'{path}/{filename}')
+        for directory in directories:
+            _log.info(f'Deleting directory {path}/{directory}...')
+            delete_directory(f'{path}/{directory}')
+            default_storage.delete(f'{path}/{directory}')
+
+    datafile_path = instance.datafile.name
+    squeezed_datafile_path = instance.datafile.name
```
Comment (mcrot) [inline topobank/manager/signals.py:62]: Should this be `instance.squeezed_datafile.name`?

File: topobank/manager/signals.py [hunk 2]
```diff
@@ -47,29 +47,46 @@ def delete_datafile(datafile_attr_name):
...
+    # Otherwise we abort deletion.
+    if datafile_path is not None and not datafile_path.startswith(instance.storage_prefix):
+        _log.warning(f'Datafile is stored at location {datafile_path}, but storage prefix is '
+                     f'{instance.storage_prefix}. I will not attempt to deleting everything at this prefix.')
+        return
+    if squeezed_datafile_path is not None and not squeezed_datafile_path.startswith(instance.storage_prefix):
+        _log.warning(f'Squeezed datafile is stored at location {squeezed_datafile_path}, but storage prefix is '
+                     f'{instance.storage_prefix}. I will not attempt to deleting everything at this prefix.')
+        return
+    if datafile_path is not None and not datafile_path.startswith(instance.storage_prefix):
```
Comment (mcrot) [inline topobank/manager/signals.py:81]: Why has this been doubled?

**Review by mcrot — CHANGES_REQUESTED** [review]
Please review my questions on `signals.py`.

File: topobank/manager/signals.py (hunk 2 above)
Comment (pastewka) [reply topobank/manager/signals.py:81]: Can be removed

**Review by mcrot — APPROVED** [review]

**Pull request merged**